src/test/store/js-tracer.test.js

```diff
@@ -0,0 +1,170 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+import { storeWithProfile } from '../fixtures/stores';
+import { selectedThreadSelectors } from '../../reducers/profile-view';
+import { ensureExists } from '../../utils/flow';
+import { changeShowJsTracerSummary } from '../../actions/profile-view';
+import {
+  getProfileFromTextSamples,
+  getProfileWithJsTracerEvents,
+  type TestDefinedJsTracerEvent,
+} from '../fixtures/profiles/make-profile';
+
+describe('selectors/getJsTracerTiming', function() {
+  describe('full stack-based view', function() {
+    it('has no JS tracer timing if no js tracer info is present', function() {
+      const { profile } = getProfileFromTextSamples('A');
+      const { getState } = storeWithProfile(profile);
+      expect(profile.threads[0].jsTracer).toBe(undefined);
+      expect(
+        selectedThreadSelectors.getExpensiveJsTracerTiming(getState())
+      ).toEqual(null);
+    });
+
+    it('has empty JS tracer timing if no events are in the js tracer table', function() {
+      expect(
+        getHumanReadableJsTracerTiming({ selfTime: false, events: [] })
+      ).toEqual([]);
+    });
+
+    it('can generate some simple nested timing', function() {
+      expect(
+        getHumanReadableJsTracerTiming({
+          selfTime: false,
+          events: [
+            ['https://mozilla.org', 0, 20],
+            ['Interpreter', 1, 19],
+            ['IonMonkey', 2, 18],
+          ],
+        })
+      ).toEqual([
+        'https://mozilla.org (0:20)',
+        'Interpreter (1:19)',
+        'IonMonkey (2:18)',
+      ]);
```

Comment:
Same remark as in other tests: we could likely use `toMatchInlineSnapshot` for the expecations in this file, which would also reduce the amount of duplication we'd have to type.